### PR TITLE
update widget by id

### PR DESCRIPTION
**Pull request by vishalvora**
to update the widget by id 

```python
# update widget
ds.write('this is text', id = 'x') 
ds.write('this is updated text', id = 'x')
```

this will print "this is updated text" only in webapp

### CODE REVIEW

File: lib/datastack/server/ds_class.py [hunk 1]
```diff
@@ -262,7 +264,7 @@ def html(self, html, id=''):
         }
         self.append_block(block)
 
-    def editable_html(self, key):
+    def editable_html(self, key, id=''):
         default_html = [
       {
         "_id": "5f54d75b114c6d176d7e9765",
```
Comment (mayurpokiya) [inline lib/datastack/server/ds_class.py:270]: is this required?

File: lib/datastack/server/ds_class.py [hunk 2]
```diff
@@ -534,9 +536,24 @@ def set_page(self, page_name):
     def append_block(self,block, location='main_page'):
         self.blocks[location].append(block)
 
-    def get_block_by_id(self, id):
+    def get_app_block_by_id(self, id):
+        print('getting block by id:', id)
```
Comment (mayurpokiya) [inline lib/datastack/server/ds_class.py:540]: can we remove this print statement ?

File: lib/datastack/server/ds_class.py (hunk 1 above)
Comment (vishalvora) [reply lib/datastack/server/ds_class.py:270]: It is standards that we put id to all widgets including editable html

File: lib/datastack/server/ds_class.py (hunk 2 above)
Comment (vishalvora) [reply lib/datastack/server/ds_class.py:540]: Yes will remove print statement

**Comment by Karan2311** [comment]
If the widget's value is stored in a variable(like input/select), then that variable's value will get updated when there is second widget variable with the same Id. Is that correct? If so then variable names should also be same, right? 

What will be the case for this example:
A = df.select(['a','b','c'],id = 'test')  # a is selected
B = df.select(['a','b','c'],id = 'test')  # b is selected

I am not getting how this will be helpful in case of widgets for which values are stored in a variable. For widgets like text/info/error which are not being stored in any variable, I got it from your example.

**Review by mayurpokiya — APPROVED** [review]

**Pull request merged**